### DISCUSSION TITLE
Persisting files and terminal sessions between browsers

### DISCUSSION THREAD

**Discussion by kzvi** [gitpod-io/openvscode-server#324] [Q&A]
Hi,

I'm interested in using this project to enable me to work from multiple computers.

It seems that if I open files or terminal tabs on one computer, and then close that browser window, and then go to a different computer and access vs code at the same url, the files/terminals are not there anymore.

But they are still running if I access from the first computer.

Is it possible to have one "session" that can be accessed across multiple computers/browsers?

Also... if I access vs code from a browser, and then clear the cookies/cache on the browser, is it possible to then access or at least somehow close any terminal sessions that were open?

**Comment by jeanp413** [comment]
Hi @kzvi 

> Is it possible to have one "session" that can be accessed across multiple computers/browsers?

Not possible without extensions, session data is stored in the browser (for opened files, regarding terminals see below). Maybe possible if you write an extension (or maybe there's one already?) and store a list of opened files per project on disk, and then use it to restore opened files in another session.

> if I access vs code from a browser, and then clear the cookies/cache on the browser, is it possible to then access or at least somehow close any terminal sessions that were open?

Good question, i just tried this and it seems it's possible, first you need to run `Terminal: Detach session` command and then `Terminal: Attach to session` command, it will show you a list of terminal to attach (I think it shouldn't be necessary to call detach session, this seems like a bug, will report it upstream)
